metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -266,6 +230,51 @@ private static CreatePublicationRequest requestWithTags(List<String> tags) {
                  );
     }
 
+    private static Stream<Arguments> provideMetadataForIdentifier() {
+        String doiWithPrefix = "doi:10.1/url";
+        String doiWithoutPrefix = "10.1/url";
+        String notADoi = "identifier/not/a/doi";
+        String expectedDoi = "https://doi.org/10.1/url";
+        String dummyTitle = "To avoid NPE in JsonLdProcessor";
+
+        CreatePublicationRequest request = requestWithIdentifier(URI.create(expectedDoi), dummyTitle);
+        CreatePublicationRequest emptyRequest = requestWithIdentifier(null, dummyTitle);
+
+        return Stream.of(
+                generateMetadataHtml(Map.of(
+                        DC_TITLE, dummyTitle,
+                        DC_IDENTIFIER, doiWithPrefix),
+                        request),
+                generateMetadataHtml(Map.of(
+                        DC_TITLE, dummyTitle,
+                        DC_IDENTIFIER_UPPER_CASE, doiWithoutPrefix),
+                        request),
+                generateMetadataHtml(Map.of(
+                        DC_TITLE, dummyTitle,
+                        DC_IDENTIFIER_UPPER_CASE, expectedDoi),
+                        request),
+                generateMetadataHtml(Map.of(
+                        DC_TITLE, dummyTitle,
+                        DC_IDENTIFIER, notADoi),
+                        emptyRequest)
+                );
+    }
+
+    private static CreatePublicationRequest requestWithIdentifier(URI identifier, String title) {
```

Comment:
These test/function names; could they be more descriptive about what they actually do? Even createRequestWithIdentifier would be an improvement imo, but perhaps the team has some standardized naming scheme?